imp_developer.py

```diff
@@ -272,56 +326,414 @@ def __base64_encode(str):
+        if code in [401]:
+            errors = response.get("errors")
+            if errors and len(errors) > 0:
+                # it is possible to get more than one error
+                # but usually it is only one error
+                for error in errors:
+                    if error.get("title") in ["'Invalid Credentials'"] or error.get("code") == "PX100":
+                        return True
+        return False
+
+    @staticmethod
+    def is_wrong_input(code, response):
+        if code in [400, 409]:
+            errors = response.get("errors")
+            if errors and len(errors) > 0:
+                # it is possible to get more than one error
+                # but usually it is only one error
+                for error in errors:
+                    if error.get("detail"):
+                        return error.get("detail"), error.get("title")
+        return None, None
+
+    @staticmethod
+    def is_failure_request(response, code):
+        failure = None
+        if not HTTP.is_response_code_valid(code):
+            if code == 404:
+                return "\n There is no Internet connection.\n Or requested resource not avialble."
+            errors = response.get("errors")
+            if errors and len(errors) > 0:
+                # take only first error to show
+                failure = errors[0]["detail"]
+        # return the failure message
+        return failure
+
+class ImpRequest():
+    INVALID_CREDENTIALS = 1
+    WRONG_INPUT = 2
+    COMPILE_FAIL = 3
+    FAILURE = 4
+
+class ImpCentral:
+
+    @staticmethod
+    def auth(user_name, password):
+        url = PL_IMPCENTRAL_API_URL_V5 + "auth"
+        response, code = HTTP.post(None, url,
+            '{"id": "' + user_name + '", "password": "' + password + '"}',
+            headers=HttpHeaders.AUTH)
+
+        error = ImpCentral.handle_http_response(response, code)
+        return response, error
+
+    @staticmethod
+    def refresh_access_token(refresh_token):
+        response, code = HTTP.post(None,
+            PL_IMPCENTRAL_API_URL_V5 + "/auth/token",
+            '{"token":"'+refresh_token+'"}',
+            headers=HttpHeaders.AUTH)
+        error = ImpCentral.handle_http_response(response, code)
+        return response, error
+
+    @staticmethod
+    def account(token):
+        response, code = HTTP.get(token,
+            PL_IMPCENTRAL_API_URL_V5 + "/accounts/me")
+        error = ImpCentral.handle_http_response(response, code)
+        return response.get("data"), error
+
+
+    @staticmethod
+    def list_products(token, owner_id=None):
+        ImpCentral.account(token)
+        # TODO: list all products
+        if owner_id:
+            data='{"filter[owner.id]:"' + owner_id + '"}'
+        else:
+            data=None
+
+        link = PL_IMPCENTRAL_API_URL_V5 + "products"
+        products = []
+
+        while link != None:
+            response, code = HTTP.get(token, link, data=data)
```

Comment:
Blocking HTTP calls is always a bad idea!

Reply:
added comments with todo for a future